File: sdk/src/main/java/com/meniga/sdk/models/accounts/MenigaAccountType.java
package com.meniga.sdk.models.accounts;

import android.os.Parcel;
import android.os.Parcelable;

import java.io.Serializable;

/**
 * Copyright 2017 Meniga Iceland Inc.
 * Represents further data on the type of account.
 */
public class MenigaAccountType implements Parcelable, Serializable {
	public static final Creator<MenigaAccountType> CREATOR = new Creator<MenigaAccountType>() {
		@Override
		public MenigaAccountType createFromParcel(Parcel source) {
			return new MenigaAccountType(source);
		}

		@Override
		public MenigaAccountType[] newArray(int size) {
			return new MenigaAccountType[size];
		}
	};

	private long id;
	private String name;
	private long parentId;
	private String parentName;

	protected MenigaAccountType(Parcel in) {
		this.id = in.readLong();
		this.name = in.readString();
		this.parentId = in.readLong();
		this.parentName = in.readString();
	}

	/**
	 * @return The ID of the account type.
	 */
	public long getId() {
		return this.id;
	}

	/**
	 * @return The name of the account type.
	 */
	public String getName() {
		return this.name;
	}

	/**
	 * @return The parent ID. Not used.
	 */
	public long getParentId() {
		return this.parentId;
	}

	/**
	 * @return The parent name. Not used.
	 */
	public String getParentName() {
		return this.parentName;
	}

	@Override
	public int describeContents() {
		return 0;
	}

	@Override
	public void writeToParcel(Parcel dest, int flags) {
		dest.writeLong(this.id);
		dest.writeString(this.name);
		dest.writeLong(this.parentId);
		dest.writeString(this.parentName);
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) {
			return true;
		}
		if (o == null || getClass() != o.getClass()) {
			return false;
		}

		MenigaAccountType that = (MenigaAccountType) o;

		if (id != that.id) {
			return false;
		}
		if (parentId != that.parentId) {
			return false;
		}
		if (name != null ? !name.equals(that.name) : that.name != null) {
			return false;
		}
		return parentName != null ? parentName.equals(that.parentName) : that.parentName == null;
	}

	@Override
	public int hashCode() {
		int result = (int) (id ^ (id >>> 32));
		result = 31 * result + (name != null ? name.hashCode() : 0);
		result = 31 * result + (int) (parentId ^ (parentId >>> 32));
		result = 31 * result + (parentName != null ? parentName.hashCode() : 0);
		return result;
	}
}

File: sdk/src/main/java/com/meniga/sdk/models/accounts/MenigaAccountType.kt
/*
 * Copyright 2017-2018 Meniga Iceland Inc.
 */
package com.meniga.sdk.models.accounts

import android.os.Parcelable
import com.meniga.sdk.models.accounts.enums.AccountCategory
import kotlinx.android.parcel.Parcelize
import java.io.Serializable

/**
 * Represents further data on the type of account.
 *
 * @param id The unique Id for the account type.
 * @param name The name of the account type.
 * @param description The description of the account type.
 * @param accountCategory he account category for this account type.
 * @param organizationId  The Id of the organization associated with this account type.
 * @param realmId The Id of the realm used to synchronize transactions.
 * @param accountCategoryDetails Extra information about the account category, for example "visa" or "amex" for the "Credit" account category.
 * @param isCashbackEnabled True if cashback can be calculated for the account.
 */
@Parcelize
data class MenigaAccountType(
        val id: Long,
        val name: String?,
        val description: String?,
        val accountCategory: AccountCategory?,
        val organizationId: Long?,
        val realmId: Long?,
        val accountCategoryDetails: String?,
        val isCashbackEnabled: Boolean
) : Parcelable, Serializable
